es-bs-base/scss/mixins/_buttons.scss

```diff
@@ -3,6 +3,141 @@
 // Easily pump out default styles, as well as :hover, :focus, :active,
 // and disabled options for all buttons
 
+@mixin button-v2-variant($variant) {
+  $background: map-get($button-colors, $variant);
+  $border: $background;
+
+  $hover-background: map-get($button-hover-colors, $variant);
+  $hover-border: map-get($button-hover-colors, $variant);
+
+  $active-background: map-get($button-active-colors, $variant);
+  $active-border: map-get($button-active-colors, $variant);
+
+  $focus-background: map-get($button-hover-colors, $variant);
+  $focus-border: map-get($button-focus-border-colors, $variant);
+
+  color: color-yiq($background);
+  @include gradient-bg($background);
+  border-color: $border;
+  @include box-shadow($btn-box-shadow);
+
+  @include hover() {
+    color: color-yiq($hover-background);
+    @include gradient-bg($hover-background);
+    border-color: $hover-border;
+  }
+
+  &:focus,
+  &.focus {
+    color: color-yiq($focus-background);
+    @include gradient-bg($focus-background);
+    border-color: $focus-border;
+    @if $enable-shadows {
+      @include box-shadow($btn-box-shadow, 0 0 0 $btn-focus-width rgba(mix(color-yiq($background), $border, 15%), .5));
+    } @else {
+      // Avoid using mixin so we can pass custom focus shadow properly
+      box-shadow: 0 0 0 $btn-focus-width rgba(mix(color-yiq($background), $border, 15%), .5);
+    }
+  }
+
+  // Disabled comes first so active can properly restyle
+  &.disabled,
+  &:disabled {
+    color: $btn-disabled-font-color;
+    background-color: $btn-disabled-bg-color;
+    border-color: $btn-disabled-bg-color;
+    // Remove CSS gradients if they're enabled
+    @if $enable-gradients {
+      background-image: none;
+    }
+  }
+
+  &:not(:disabled):not(.disabled):active,
```

Comment:
Ok so `:not(:disabled, .disabled)` is like `!A OR !B`, and `:not():not()` is like `!A AND !B`. So this is not disabled, and active. I wonder if a comment like "active & not disabled state" could be helpful.

Reply:
Fair point! Added.